### PR TITLE
Add a simple abstract frame navigation interface

### LINKED ISSUE #118: Build a frame navigation interface (Library 3)
## User story
1. As a developer
2. I want / need an API 
3. So that Library 3 can use functionality of Library 2. 

## Acceptance criteria
* It should be implemented as an abstract class or interface
* Jump n frames forward or backward 
* Jump to a specified percentage of the diff video 
* Jump to a concretely specified frame 
* Jump to the next diff at a specified position

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by zino212**
Closes #118 

This PR adds a simple abstract frame navigation interface for later use by the GUI (#107) and implementation in lib2 (#125).

### CODE REVIEW

File: GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+/**
+ * Provides navigation methods for frames in a video.
+ *
+ * This interface defines methods to navigate frames in various ways, including jumping a specified number
+ * of frames in a given direction, jumping to a specific percentage of the video, jumping to a specific frame,
+ * and jumping to the next difference frame. Implementing classes should provide implementation for these methods.
+ */
+interface AbstractFrameNavigationInterface {
```
Comment (akriese) [inline GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt:9]: I think, you can leave out the Abstract in the name, possibly also the Interface, but we dont really have consistent conventions atm.

File: GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    /**
+     * Jump n frames in a specified direction
+     */
+    abstract fun jumpFrames(frames: Int, forward: Boolean)
```
Comment (akriese) [inline GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt:14]: The abstract does not have a purpose here, as this is now an interface and by definition (implicitly) all functions are abstract anyways :)

**Review by akriese — CHANGES_REQUESTED** [review]

File: GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt (hunk 1 above)
Comment (zino212) [reply GUI/src/main/kotlin/interface/AbstractFrameNavigationInterface.kt]: I left Interface in the title. I like to see the purpose/type of the class also in the filename and I think per convention the class has to have the same name as its file.

**Comment by akriese** [comment]
Before merging this, lets wait for other devs' opinions about this. Otherwise, LGTM

File: GUI/src/main/kotlin/interface/FrameNavigationInterface.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * This interface defines methods to navigate frames in various ways, including jumping a specified number
+ * of frames in a given direction, jumping to a specific percentage of the video, jumping to a specific frame,
+ * and jumping to the next difference frame. Implementing classes should provide implementation for these methods.
+ */
+interface FrameNavigationInterface {
+
+    /**
+     * Jump n frames in a specified direction
+     */
+    fun jumpFrames(frames: Int, forward: Boolean)
```
Comment (a-miscellaneous) [inline GUI/src/main/kotlin/interface/FrameNavigationInterface.kt:14]: boolean is unnecessary, since frames can be negative

Comment (zino212) [reply GUI/src/main/kotlin/interface/FrameNavigationInterface.kt:14]: I thought that an additional boolean would be more clear, but sure I can remove it.

**Pull request merged**